### ISSUE TITLE
Cannot export hook because QueryHook is a private interface

### ISSUE THREAD

**Issue by MatthewCushing** [liaoliao666/react-query-kit#3]
Trying to do something like:

```typescript
import { createQuery } from 'react-query-kit';

export const useUsers = createQuery<Response, Variables, Error>(queryOptions);
```

results in a typescript error of 

```shell
Exported variable 'useUsers' has or is using name 'QueryHook' from external module "path/to/project/node_modules/.pnpm/react-query-kit@1.2.10_djz4nnd5briyzzf typescript (4023)
```

Exporting QueryHook from the library allows me to:

```typescript
import type { QueryHook } from '../../../node_modules/react-query-kit/src/index';
import { createQuery } from 'react-query-kit';

export const useUsers: QueryHook<Response, Variables, Error> = createQuery<Response, Variables, Error>(queryOptions);
```

That's a solution but it would be nicer if typescript could infer the type for `useUsers`.

**Comment by liaoliao666** [comment]
could u provide a sandbox example to me?
I can not reproduce this case, cuz the type for cutom hook that work to me fine

**Comment by liaoliao666** [comment]
Or you are using  `react-query` instead of `@tanstack/react-query`, which also causes this problem

**Comment by LeeWeeLong** [comment]
I encounter the same issue. May I know is it possible to export the QueryHook interface?

**Comment by liaoliao666** [comment]
@MatthewCushing 
Exporting the Query Hook interface is not the best solution. Becuase of the created hook will infer the type automatically.
It is not a good experience to importing the QueryHook interface for Type Inference.

Could u provide a codesandbox to me to find out what is causing the problem.

**Comment by liaoliao666** [comment]
@LeeWeeLong 
Thanks for your work very much.
It will be fixed in the next version, just wait for a few hours

**Comment by liaoliao666** [comment]
@LeeWeeLong @MatthewCushing 
Hey guys. It has been fixed in the latest version 1.2.11. Just enjoy it.
